saleor/tests/e2e/checkout/discounts/sales/test_checkout_product_on_percentage_sale.py

```diff
@@ -113,47 +166,225 @@ def test_checkout_products_on_percentage_sale_core_1004(
         set_default_shipping_address=True,
     )
     checkout_id = checkout_data["id"]
-    checkout_lines = checkout_data["lines"][0]
-    line_discount = round(float(product_variant_price) * sale_discount_value / 100, 2)
-    unit_price = float(product_variant_price) - line_discount
+    checkout_line1 = checkout_data["lines"][0]
+    checkout_line2 = checkout_data["lines"][1]
+
+    assert checkout_line1["variant"]["id"] == product1_variant_id
+    assert checkout_line1["quantity"] == product1_quantity
+    line1_discount = round(product1_variant_price * sale_discount_value / 100, 2)
+    line1_unit_price = product1_variant_price - line1_discount
+    assert checkout_line1["unitPrice"]["gross"]["amount"] == line1_unit_price
```

Comment:
I think we should also check line total prices, in all places, where the `unitPrice` is checked